src/v2/Components/Artwork/SaveButton/index.tsx

```diff
@@ -1,234 +1,204 @@
   }
 
-  renderCustomButton() {
+  if (render) {
     return (
-      <div {...this.mixinButtonActions()} data-test="saveButton">
-        {/* @ts-expect-error STRICT_NULL_CHECK */}
-        {this.props.render(this.props, this.state)}
-      </div>
+      <Clickable
+        onClick={handleClick}
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}
+        data-test="saveButton"
+      >
+        {render(props, state)}
+      </Clickable>
     )
   }
 
-  render() {
-    return this.props.render
-      ? this.renderCustomButton()
-      : this.renderDefaultButton()
-  }
+  return (
+    <Container
+      data-test="saveButton"
+      position="absolute"
+      right={10}
+      bottom={10}
+      isSaved={isSaved}
+      onClick={handleClick}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <Inner isSaved={isSaved}>
+        {isRemoveable ? (
+          <CloseIcon fill="white100" width={24} height={24} />
+        ) : (
+          <HeartIcon fill="white100" width={24} height={24} />
+        )}
+      </Inner>
+    </Container>
+  )
 }
 
-export const ClickContainer = styled(Clickable)<{ isSaved: boolean }>`
+export const Container = styled(Clickable)<{ isSaved: boolean }>`
   position: absolute;
-  right: ${space(1)}px;
-  bottom: ${space(1)}px;
+  right: ${themeGet("space.1")};
+  bottom: ${themeGet("space.1")};
   opacity: ${({ isSaved }) => (isSaved ? 1 : 0)};
 `
 
-export const Container = styled(Flex)<{ isSaved: boolean }>`
-  width: ${SIZE}px;
-  height: ${SIZE}px;
-  background-color: ${({ isSaved }) =>
-    isSaved ? themeGet("colors.brand") : "rgba(0, 0, 0, 0.4)"};
+export const Inner = styled(Flex)<{ isSaved: boolean }>`
+  width: 40px;
+  height: 40px;
   border-radius: 50%;
   justify-content: center;
   align-items: center;
 
-  &:hover {
-    background-color: ${({ isSaved }) =>
-      isSaved ? color("red100") : color("black100")};
-  }
+  ${({ isSaved }) => {
+    return isSaved
+      ? css`
+          background-color: ${themeGet("colors.brand")};
+          &:hover {
+            background-color: ${themeGet("colors.red100")};
+          }
+        `
+      : css`
+          background-color: rgba(0, 0, 0, 0.4);
+          &:hover {
+            background-color: ${themeGet("colors.black100")};
+          }
+        `
+  }}
 `
 
-export const SaveButtonFragmentContainer = createFragmentContainer(
-  Artsy.withSystemContext(SaveButton) as React.ComponentType<SaveButtonProps>,
-  {
-    artwork: graphql`
-      fragment SaveButton_artwork on Artwork {
-        id
-        internalID
-        slug
-        is_saved: isSaved
-        title
-      }
-    `,
-  }
-)
+export const SaveButtonFragmentContainer = createFragmentContainer(SaveButton, {
+  artwork: graphql`
+    fragment SaveButton_artwork on Artwork {
+      id
+      internalID
+      slug
+      is_saved: isSaved
+      title
+    }
+  `,
+})
```

Comment:
This was just a mindless translation to functional component with the addition of using `useSystemContext` instead of accepting mediator/user as props. We could clean this up a lot but save that for another day.